docs/oathkeeper/guides/websockets.mdx

```diff
@@ -0,0 +1,360 @@
+  kratos-selfservice-ui-node:
+    image: oryd/kratos-selfservice-ui-node:latest
+    environment:
+      - KRATOS_PUBLIC_URL=http://kratos:4433/
+      - KRATOS_BROWSER_URL=http://127.0.0.1:4433/
+    networks:
+      - intranet
+    ports:
+      - "4455:3000"
+    restart: on-failure
+  mailslurper:
+    image: oryd/mailslurper:latest-smtps
+    ports:
+      - '4436:4436'
+      - '4437:4437'
+    networks:
+      - intranet
+networks:
+  intranet:`}</CodeBlock>
+
+This example uses the following network architecture:
+
+- `4433` port is a public ("Browser") API of Ory Kratos
+- `4434` is an admin API of Ory Kratos
+- `4455` is a port of the User Interface implemented by self-service-ui
+- `8080` is a port of Ory Oathkeeper
+
+The rest ports or services will be available only inside the internal network.
+
+## Configuring Ory Oathkeeper and Ory Kratos
+
+1. Let's create a `kratos` folder and fetch configuration files
+
+<CodeBlock className="language-shell">{`mkdir kratos
+wget https://raw.githubusercontent.com/ory/kratos/${useLatestRelease('kratos')}/contrib/quickstart/kratos/email-password/identity.schema.json -O kratos/identity.schema.json
+wget https://raw.githubusercontent.com/ory/kratos/${useLatestRelease('kratos')}/contrib/quickstart/kratos/email-password/kratos.yml -O kratos/kratos.yml`}</CodeBlock>
+
+2. Create `oathkeeper` folder and `oathkeeper/oathkeeper.yml` with the following content
+
+```yaml
+log:
+  level: debug
+  format: json
+
+serve:
+  proxy:
+    cors:
+      enabled: true
+      allowed_origins:
+        - http://127.0.0.1:8080
+      allowed_methods:
+        - POST
+        - GET
+        - PUT
+        - PATCH
+        - DELETE
+      allowed_headers:
+        - Authorization
+        - Content-Type
+      exposed_headers:
+        - Content-Type
+      allow_credentials: true
+      debug: true
+
+errors:
+  fallback:
+    - json
+
+  handlers:
+    redirect:
+      enabled: true
+      config:
+        to: http://127.0.0.1:4455/login
+        when:
+          -
+            error:
+              - unauthorized
+              - forbidden
+            request:
+              header:
+                accept:
+                  - text/html
+    json:
+      enabled: true
+      config:
+        verbose: true
+
+access_rules:
+  matching_strategy: glob
+  repositories:
+    - file:///etc/config/oathkeeper/access-rules.yml
+
+authenticators:
+  anonymous:
+    enabled: true
+    config:
+      subject: guest
+
+  cookie_session:
+    enabled: true
+    config:
+      check_session_url: http://kratos:4433/sessions/whoami
+      preserve_path: true
+      extra_from: "@this"
+      subject_from: "identity.id"
+      only:
+        - ory_kratos_session
+
+  noop:
+    enabled: true
+
+authorizers:
+  allow:
+    enabled: true
+
+mutators:
+  noop:
+    enabled: true
+```
+
+3. Create `oathkeeper/access-rules.yml` with the following content
+
+```yaml
+-
```

Comment:
should this `-` be here?

Reply:
Yes. That's the configuration of `access-rules.yml` and it has an array of elements; hence we need to have this `-` sign.